server/controllers/profile.js

```diff
@@ -0,0 +1,69 @@
+const mongoose = require('mongoose');
+const Profile = require('../models/Profile');
+const asyncHandler = require('express-async-handler');
+
+// POST /profile - Create new profile
+exports.createNewProfile = asyncHandler(async (req, res) => {
+  const profile = req.body;
+  const newProfile = new Profile({ ...profile });
+
+  if (!newProfile) {
+    res.status(500);
+    throw new Error('Error creating the profile.');
+  }
+  await newProfile.save();
+  res.status(201).json({ status: 201, profile: newProfile, message: 'Profile created!' });
+});
+
+// UPDATE /profile:id - Update profile with the given id
+exports.updateProfile = asyncHandler(async (req, res) => {
+  const id = req.params.id;
+  const profile = req.body;
+
+  try {
+    const updatedProfile = await Profile.findByIdAndUpdate(id, profile, {
+      new: true,
+    });
+    res.status(201).json({ status: 201, profile: updatedProfile, message: 'Profile updated!' });
+  } catch (error) {
+    res.status(500);
+    throw new Error('Error updating the profile.');
+  }
+});
+
+// GET /profile:id - Search profile with the given id
+exports.searchProfiles = asyncHandler(async (req, res) => {
```

Comment:
I guess this is a typo - searchProfile here, as we are looking for a specific profile :)